lib/redmine/menu_manager/wiki_menu_helper.rb

```diff
@@ -67,4 +73,32 @@ def push_wiki_menu_subitem(menu, main_item, child)
     Rails.logger.error "Failed to add wiki item #{child.slug} to wiki menu: #{e}. Deleting it."
     child.destroy
   end
+
+  def default_menu_item(page)
+    if (main_item = page.nearest_main_item)
+      main_item
+    else
+      MenuItems::WikiMenuItem.main_items(page.wiki.id).first
+    end
+  end
+
+  private
+
+  def push_wiki_menu_partial(main_item, menu)
+    menu.push :wiki_menu_partial,
+              { controller: '/wiki', action: 'show' },
+              param: :project_id,
+              parent: main_item.menu_identifier,
+              partial: 'wiki/menu_pages_tree',
+              last: true
+  end
+
+  def push_wiki_entry_page_item(main_item, menu)
+    menu.push "entry-item-#{main_item.menu_identifier}".to_sym,
```

Comment:
You may want to extract `"entry-item-#{item.menu_identifier}".to_sym` into a separate helper method since it occurs in many places